### PR TITLE
fix: 🐛 Fixes sourcing secrets file - #517

### LINKED ISSUE #517: envalid.ts contains bug that doesn't parse .secrets properly
**Describe the bug:** (a clear and concise description of what the bug is)
if `ENV_DIR/.secrets` contains `export FOO=BAR`, then `src/common/envalid.ts` won't parse FOO=BAR, because of the export

**To Reproduce**

1. Create a .secrets file with exported variables
2. Print `env` in playground
3. Observe that FOO is not parsed

Alernative:
1. have ENV_DIR point to valid values
2. run `binzx/otomi lint` 
3. Observe templating issues with `requiredEnv: required env var GCLOUD_SERVICE_KEY`

**Expected behavior:** (a clear and concise description of what you expected to happen)
FOO being parsed

**Proposed solution:**
Environment variables from a file (like .secrets or .env) are reflected both in `process.env` as well as within the zx shell commands

### PR DESCRIPTION
**Pull request by Dunky13**
This PR fixes #517 
We cannot rely on just dotenv, since it cannot & will not parse lines that start with `export `
This change works around it :)

### CODE REVIEW

File: src/common/envalid.ts [hunk 1]
```diff
@@ -1,6 +1,22 @@
-import { config } from 'dotenv'
+import { DotenvParseOutput, parse } from 'dotenv'
 import { bool, cleanEnv, json, str } from 'envalid'
-import { existsSync } from 'fs'
+import { existsSync, readFileSync } from 'fs'
+
+export const dotEnvParse = (path: string): DotenvParseOutput => {
```
Comment (j-zimnowoda) [inline src/common/envalid.ts:5]: I think that this function should not update process.env 
The single responsibility is to parse env file and return parsed content

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]

File: src/common/envalid.ts [hunk 2]
```diff
@@ -15,11 +31,9 @@ const cleanSpec = {
...
 }
-let pEnv: any = process.env
-const path = `${pEnv.ENV_DIR}/.secrets`
-if (pEnv.ENV_DIR && existsSync(path)) {
-  const result = config({ path }) // this sets vars from .env onto process.env
-  if (result.error) console.error(result.error)
-  pEnv = { ...pEnv, ...result.parsed }
+
+const path = `${process.env.ENV_DIR}/.secrets`
+if (process.env.ENV_DIR && existsSync(path)) {
```
Comment (j-zimnowoda) [inline src/common/envalid.ts:36]: add comment that you do it because to introduce caching like mechanism

Comment (Dunky13) [reply src/common/envalid.ts:36]: It doesn't?

File: src/common/envalid.ts (hunk 1 above)
Comment (Dunky13) [reply src/common/envalid.ts:5]: Isn't it desirable to add it to `process.env`?

**Comment by Morriz** [comment]
Why is this PR created? What was wrong with the setup before? We don't need `export` in env files.

**Comment by Morriz** [comment]
Please consult with me before making such assumptions next time

File: src/common/envalid.ts (hunk 2 above)
Comment (Dunky13) [reply src/common/envalid.ts:36]: It just checks if the files are present, otherwise skip.

**Comment by Morriz** [comment]
Closing as unproductive as new format without export works as intended (also works now with docker compose). Please update your secrets files as such.

**Pull request closed without merging**